### DISCUSSION TITLE
Branch preview URL / Domain not deploying

### DISCUSSION THREAD

**Discussion by michedapt** [vercel/community#3202] [Help] [answered]
### Summary

Hello. 

I'm trying to make a deployment via a branch I have in github. 

When I open up a PR, the deployment to preview starts up as normal. However, it appears as if the preview branch url / domain is not deploying at all in the platform. I haven't had this issue before. 

### Steps to Reproduce

Here's an example of one of my normal deployments to preview (branch A):
![gooddeploy](https://github.com/vercel/community/assets/132643406/995c5cd6-df88-471f-8c8e-612ffdd66a8c)

Here's an example of my current branch that is failing (branch B):
![baddeploy](https://github.com/vercel/community/assets/132643406/81e76ccb-3bc1-4af0-bd47-9e523f3a53ac)

As you can see, Branch B does not have the branch URL / domain. It also fails to build. 
From the deployment logs it appears as if one of my pages tries to prerender, which requires a fetch to /api/. This requires the preview branch URL / Domain and the /api/ to be deployed there?. So I am getting 404s on that fetch.

NOTE: From within the code, I can actually get the VERCEL_BRANCH_URL environment variable. I've logged that environment variable and it is correct but why doesn't it show on the vercel deployment UI? 

Does anyone know what I should do here? Any help will be appreciated.

I've seen a similar issue on stack:

https://stackoverflow.com/questions/71655859/vercel-automatic-branch-url-not-deploying

**Comment by amyegan** [comment, marked as answer]
Hi @michedapt. The build failure is why the branch URL is not being applied to the deployment. The branch URL is supposed to stay up to date with the latest successful deployment. In this case nothing was actually deployed because the build failed.

If you switch the prerendered page to SSR or change the code so it does not call the `/api` route which is not yet deployed at build time, that should solve the build error. Once you get a new successful deployment, the branch URL will be applied to it.

**Reply by michedapt** [reply]
Hi @amyegan. Thanks for the quick response. I understand now that this domain is not going to be available until a successful build is created.

I do have a question regarding prerendering if that's okay. 

I'm using the app router in next 13. From what I read, every page unless marked with 'use client' at the top is going to be SSR by default. Which is the case with my page where I am fetching data. 

Here's an example of that page that is causing the issue:

It's located in:

app/(dashboard)/Integrations/page.tsx

```
import IntegrationsList from "./components/integrations-list"

import axios from "axios"

async function getData() {
  let host
  if (process.env.VERCEL_ENV == "development") {
    host = "http://localhost:3000"
  } else if (process.env.VERCEL_ENV == "preview") {
    host = `https://${process.env.VERCEL_BRANCH_URL}`
  } else if (process.env.VERCEL_ENV == "production") {
    host = `https://${process.env.VERCEL_URL}`
  }

  let tokenData, dbData

  try {
    const res = await axios.get(`${host}/api/token`)
    tokenData = res.data.toke.paragonToken as string
  } catch (error) {
    console.error("Failed to fetch token data:", error)
    console.error("host:", host)
    // This will activate the closest `error.js` Error Boundary
    throw new Error("Failed to fetch token data")
  }

  try {
    const resDB = await axios.post(`${host}/api/team`, { title: "React POST Request Example" }, { headers: { "Content-Type": "application/json" } })
    dbData = resDB.data.team_info
  } catch (error) {
    console.error("Failed to fetch team data:", error)
    console.error("host:", host)
    // This will activate the closest `error.js` Error Boundary
    throw new Error("Failed to fetch team data")
  }

  return { team: dbData, token: tokenData }
}

export default async function Page() {
  const data = await getData()
  
  return <IntegrationsList {...data} />
  
}
```

If the page is SSR (as shown above) then vercel should not need to prerender, but it appears to do so still.

**Reply by smaeda-ks** [reply]
you can change the behavior:
https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config

for example, adding `export const dynamic = 'auto'` would make it fully dynamic

**Reply by michedapt** [reply]
Thanks for that. @smaeda-ks
